### ISSUE TITLE
The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior.

### ISSUE THREAD

**Issue by zhuxiaofan117** [manycore-research/SpatialLM#21]
> 你好，我怀疑可能是生成的时候timeout了 你可以试着增大timeout的值 在这一行代码中 [line81](https://github.com/manycore-research/SpatialLM/blob/02d7e6e11db127483117440440805356c86bb222/inference.py#L81).
> 
> 或者也可以尝试一下删除timeout，这样会取消timeout机制。 

 _Originally posted by @SamMaoYS in [#6](https://github.com/manycore-research/SpatialLM/issues/6#issuecomment-2742794207)_

Hi, I noticed there has been a discussion on this warning. And as suggested, I have deleted timeout in reference.py. 
However, when I am running evaluation on SpatialLM-Test dataset,  this warning "The attention mask and the pad token id were not set" still persists for every single data.

![Image](https://github.com/user-attachments/assets/a3c77954-bd6b-4758-8504-8a19f9cb1abe)

I wonder if this warning has some penalty effect on the inferenced result, since my evaluation result of **SpatialLM-Llama-1B** is lower than the official Benchmark released on almost every single item, as shown below.

![Image](https://github.com/user-attachments/assets/f69b4f1c-fa75-4c53-8dbb-641610eab550)

Also noticed in previous post, this warning may be caused by machine's computing capabilitiesm, my Server's installation is:
CPU: Intel(R) Xeon(R) Platinum 8462Y+  
RAM: 64G*8
GPU: NVIDIA L40 48g      Driver Version: 550.120        CUDA Version: 12.4

For conda env installation, I followed Mast3r-Slam‘s guideline(which happens to use similar CUDA and pytorch version :D) and handly pip install all poetry related packages(due to poetry's proxy issues).

Much appreciated if devs could take a look on my question. Thx in advance.

**Comment by SamMaoYS** [comment]
Hi, thank you for the question. The warning is more related to the design of Transformers library. Check out a related issue [Qwen issue](https://github.com/QwenLM/Qwen2.5/issues/26#issuecomment-2358536071). I don't think it will affect the results.

The LLM outputs do have some randomness, so it is expected the results may vary for each run. And based on the results you shared, I think it is inside the error margin. We will keep improving to come up with better benchmarking methods.
